client/state/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import tv4 from 'tv4';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * Module variables
+ */
+
+export function isValidStateWithSchema( state, schema, checkForCycles = false, banUnknownProperties = false ) {
+	const result = tv4.validateResult( state, schema, checkForCycles, banUnknownProperties );
+	if ( ! result.valid && config( 'env' ) === 'development' ) {
```

Comment:
>The only problem I ran into was that it was somewhat difficult to mock out the console.warn.bind in test.

Yeah, I might imagine that to be the case. Generally, I think we shouldn't let tests dictate how best to write the function, and the mocking is only marginally more verbose.